src/User.ts

```diff
@@ -37,6 +37,14 @@ class UserService extends Base {
     return (json as unknown) as User;
   }
 
+  async getOneByEmail({ email }: { email: string }): Promise<User> {
+    const { apiEndpointAddress, got } = this.lensPlatformClient;
+    const url = `${apiEndpointAddress}/users?email=${email}`;
+    const json = await got.get(url);
+
+    return (json as unknown) as User;
```

Comment:
This would be `User[]` actually. Should we check array length and return e.g. undefined or throw error if array length is not 1?

Reply:
Just thought about this wouldnt it make sense to just return 404 in case not found? It shouldnt be possible that multiple resulsts can be returned as emailaddress should be unique in our systems right?

Reply:
I'd prefer an empty array just to follow "RESTful practices", see e.g. https://stackoverflow.com/a/6046776

Reply:
I fully agree for collections but in this case we are only asking for some specific information...

Reply:
How would the result look like in case the user was found? Would it be an Array of Users where its always max 1 entry?

Reply:
This should be renamed as getMany with mandatory email filter, and then it would make more sense. I don't really want to make the API inconsistent in this case.

Reply:
Got your point now! you are right!